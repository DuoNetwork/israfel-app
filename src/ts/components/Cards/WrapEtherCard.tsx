import * as React from 'react';
import * as CST from '../../common/constants';
import web3Util from '../../common/web3Util';
import { SDivFlexCenter } from '../_styled';
import { SCard, SCardConversionForm, SCardList, SCardTitle, SInput } from './_styled';

interface IState {
	amount: string;
}

export default class WrapEtherCard extends React.Component<{}, IState> {
	constructor(props: object) {
		super(props);
		this.state = {
			amount: ''
		};
	}

	private handleAmountInputChange = (value: string) =>
		this.setState({
			amount: value
		});

	public render() {
		return (
			<SCard
				title={<SCardTitle>{CST.TOKEN_WETH.toUpperCase()}</SCardTitle>}
				width="370px"
				margin="0 0 0 5px"
			>
				<SCardConversionForm>
					<SCardList>
						<div className="status-list-wrapper">
							<ul>
								<li className="block-title">
									<span>{CST.TOKEN_WETH}</span>
								</li>
								<li className={'input-line'}>
									<span className="title" style={{ width: 200 }}>
										{CST.TH_AMT}
									</span>
									<SInput
										width="60%"
										className={''}
										value={this.state.amount}
										onChange={e => this.handleAmountInputChange(e.target.value)}
										placeholder={CST.TH_AMT}
										right
									/>
								</li>
								<li>
									<SDivFlexCenter
										horizontal
										width="100%"
										padding="0"
										marginTop="10px"
									>
										<button
											className={'form-button'}
											disabled={!Number(this.state.amount)}
											onClick={() => web3Util.wrapEther(Number(this.state.amount))}
										>
											{CST.TH_WRAP}
										</button>

										<button
											className={'form-button'}
											disabled={!Number(this.state.amount)}
											onClick={() => web3Util.unwrapEther(Number(this.state.amount))}
										>
											{CST.TH_UNWRAP}
										</button>
									</SDivFlexCenter>
								</li>
							</ul>
						</div>
					</SCardList>
				</SCardConversionForm>
			</SCard>
		);
	}
}
