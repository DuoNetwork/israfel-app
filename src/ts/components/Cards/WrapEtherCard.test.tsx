/**
 * @jest-environment jsdom
 */

import { shallow } from 'enzyme';
import * as React from 'react';
import WrapEtherCard from './WrapEtherCard';

describe('WrapEtherCard Test', () => {
	describe('Test Snapshot', () => {
		it('Test Snapshot', () => {
			const wrapper = shallow(
				<WrapEtherCard />
			);
			expect(wrapper).toMatchSnapshot();
			});
		});
});
