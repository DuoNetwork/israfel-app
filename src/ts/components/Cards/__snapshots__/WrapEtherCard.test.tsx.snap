// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrapEtherCard Test Test Snapshot Test Snapshot 1`] = `
<Styled(Card)
  margin="0 0 0 5px"
  title={
    <styled.div>
      WETH
    </styled.div>
  }
  width="370px"
>
  <styled.div>
    <styled.div>
      <div
        className="status-list-wrapper"
      >
        <ul>
          <li
            className="block-title"
          >
            <span>
              WETH
            </span>
          </li>
          <li
            className="input-line"
          >
            <span
              className="title"
              style={
                Object {
                  "width": 200,
                }
              }
            >
              Amount
            </span>
            <styled.input
              className=""
              onChange={[Function]}
              placeholder="Amount"
              right={true}
              value=""
              width="60%"
            />
          </li>
          <li>
            <styled.div
              horizontal={true}
              marginTop="10px"
              padding="0"
              width="100%"
            >
              <button
                className="form-button"
                disabled={true}
                onClick={[Function]}
              >
                Wrap
              </button>
              <button
                className="form-button"
                disabled={true}
                onClick={[Function]}
              >
                Unwrap
              </button>
            </styled.div>
          </li>
        </ul>
      </div>
    </styled.div>
  </styled.div>
</Styled(Card)>
`;
